src/intelligence_layer/connectors/data/data.py

```diff
@@ -0,0 +1,245 @@
+import json
+import warnings
+from collections.abc import Iterator
+from http import HTTPStatus
+from typing import Any, ClassVar
+from urllib.parse import urlencode, urljoin
+
+import requests
+from requests import HTTPError
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+
+from intelligence_layer.connectors.data.exceptions import (
+    DataError,
+    DataExternalServiceUnavailable,
+    DataForbiddenError,
+    DataInternalError,
+    DataInvalidInput,
+    DataResourceNotFound,
+)
+from intelligence_layer.connectors.data.models import (
+    DataDataset,
+    DataRepository,
+    DataRepositoryCreate,
+    DatasetCreate,
+)
+
+
+class DataClient:
+    """Data Client class to interact with the Data Platform API.
+
+    Attributes:
+        headers: headers used in the request session
+    """
+
+    _status_code_to_exception: ClassVar[dict[int, type[DataError]]] = {
+        HTTPStatus.SERVICE_UNAVAILABLE: DataExternalServiceUnavailable,
+        HTTPStatus.NOT_FOUND: DataResourceNotFound,
+        HTTPStatus.UNPROCESSABLE_ENTITY: DataInvalidInput,
+        HTTPStatus.FORBIDDEN: DataForbiddenError,
+    }
+
+    def __init__(
+        self,
+        token: str | None,
+        base_data_platform_url: str = "http://localhost:8000",
+        session: requests.Session | None = None,
+    ) -> None:
+        """Initialize the Data Client.
+
+        Args:
+            token: Access token
+            base_data_platform_url: Base URL of the Data Platform API
+            session: Requests session.
+        """
+        self._base_data_platform_url = base_data_platform_url
+        self.headers = {
+            **({"Authorization": f"Bearer {token}"} if token is not None else {}),
+        }
+
+        self._session = session or requests.Session()
+        retry_strategy = Retry(
+            total=3,  # Total number of retries
+            backoff_factor=0.5,  # Exponential backoff factor
+            status_forcelist=[
+                HTTPStatus.TOO_MANY_REQUESTS,  # 429,
+                HTTPStatus.INTERNAL_SERVER_ERROR,  # 500,
+                HTTPStatus.BAD_GATEWAY,  # 502,
+                HTTPStatus.SERVICE_UNAVAILABLE,  # 503,
+                HTTPStatus.GATEWAY_TIMEOUT,  # 504,
+            ],  # Retry on these HTTP status codes
+        )
+
+        # Mount the retry strategy to the session
+        adapter = HTTPAdapter(max_retries=retry_strategy)
+        self._session.mount("http://", adapter)
```

Comment:
do we use any http:// requests? if no, this line can go

Reply:
If we want to run(studio and it services) in a local setup we will do HTTP, right? 🤔